tools/pubsys/src/aws/validate_ami/results.rs

```diff
@@ -0,0 +1,1028 @@
+//! The results module owns the reporting of EC2 image validation results.
+
+use super::ami::{ImageDef, Result};
+use aws_sdk_ec2::Region;
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, HashSet};
+use std::fmt::{self, Display};
+use std::str::FromStr;
+use tabled::{Table, Tabled};
+
+/// Represent the possible status of an EC2 image validation
+#[derive(Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
+pub(crate) enum AmiValidationResultStatus {
+    /// The AMI was found and its monitored fields have the expected values
+    Correct,
+
+    /// The AMI was found but some of the monitored fields do not have the expected values
+    Incorrect,
+
+    /// The image was expected but not included in the actual images
+    Missing,
+}
+
+impl Display for AmiValidationResultStatus {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            AmiValidationResultStatus::Correct => write!(f, "Correct"),
+            AmiValidationResultStatus::Incorrect => write!(f, "Incorrect"),
+            AmiValidationResultStatus::Missing => write!(f, "Missing"),
+        }
+    }
+}
+
+impl FromStr for AmiValidationResultStatus {
+    type Err = super::Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        match s {
+            "Correct" => Ok(Self::Correct),
+            "Incorrect" => Ok(Self::Incorrect),
+            "Missing" => Ok(Self::Missing),
+            filter => Err(Self::Err::InvalidStatusFilter {
+                filter: filter.to_string(),
+            }),
+        }
+    }
+}
+
+/// Represents a single EC2 image validation result
+#[derive(Debug, Eq, Hash, PartialEq, Serialize)]
+pub(crate) struct AmiValidationResult {
+    /// The id of the image
+    pub(crate) id: String,
+
+    /// ImageDef containing expected values for the image
+    pub(crate) expected_image_def: ImageDef,
+
+    /// ImageDef containing actual values for the image
+    pub(crate) actual_image_def: Option<ImageDef>,
+
+    /// The region the image resides in
+    #[serde(serialize_with = "serialize_region")]
+    pub(crate) region: Region,
+
+    /// The validation status of the image
+    pub(crate) status: AmiValidationResultStatus,
```

Comment:
I think I suggested this be added as a field in a prior revision, since it's otherwise not really a `Result` being computed, but rather a validation computation being created which is later used to compute the result.

I guess if we renamed this `AmiValidation` with a method `AmiValidation::get_result()` or something, I'd be ok with that.